src/lib/parts.ts

```diff
@@ -37,6 +37,61 @@ export const isIterable = (value: unknown): value is Iterable<unknown> => {
       !!(value && (value as any)[Symbol.iterator]);
 };
 
+/**
+ * A Part that controls all or part of an attribute value.
+ */
+export class SpreadPart implements Part {
+  value: unknown = undefined;
+  subparts: Map<string, ReadonlyArray<Part>> = new Map();
+  element: Element;
+  handleAttribute: (name: string) =>  ReadonlyArray<Part>;
+  // Names of attributes that were spread in the last commit,
+  // but are not anymore.
+  private __removedNames: Array<string> = [];
+
+  constructor(element: Element, handleAttribute: (name: string) =>  ReadonlyArray<Part>) {
+    this.element = element;
+    this.handleAttribute = handleAttribute;
+  }
+
+  setValue(value: unknown): void {
+    let spreadable = value as {[key: string]: any};
+    for (var name in spreadable) {
+      this.subparts.set(name, this.handleAttribute(name))
+    }
+    for (var [name, parts] of this.subparts.entries()) {
+      if (parts) {
+        // Only one part per key is allowed via Spread
+        parts[0].setValue(spreadable[name]);
+        // Prepare to remove unused attributes
+        if (!(name in spreadable)) {
+          this.__removedNames.push(name)
+        }
+      }
+    }
+    this.value = value;
+  }
+
+  commit() {
+    for (var parts of this.subparts.values()) {
+      if (parts) {
+        // Only one part per key is allowed via Spread
+        parts[0].commit()
+      }
+    }
+    // Remove attributes that used to be spread, but aren't anymore.
+    // Removal of properties and events are already handled by committing
+    // the undefined value.
+    for (var name of this.__removedNames) {
+      let parts = this.subparts.get(name) || [];
+      if (parts && parts[0] instanceof AttributePart) {
```

Comment:
what about PropertyPart, BooleanAttributePart and EventPart?

Reply:
At first I didn't have PropertyPart here because we'd commit `undefined` while removing a property from a spread, but I just realized we should remove the property key from the element as well. 👍

Reply:
BooleanAttributePart and EventPart both handle their own clean-up when their value gets set to `undefined` by the SpreadPart. I don't love that these cleanups happen in different locations